lib/query_components/filter.rb

```diff
@@ -1,23 +1,27 @@
+require "query_components/user_filter"
+require "query_components/visibility_filter"
+require 'elasticsearch/queries'
+
 module QueryComponents
   class Filter < BaseComponent
+    include Elasticsearch::Queries
+
     def payload(excluding = [])
-      rejects = []
-      filters = []
+      user_filter = QueryComponents::UserFilter.new(search_params)
+      visibility_filter = QueryComponents::VisibilityFilter.new(search_params)
 
-      param_filters = search_params.filters.reject do |filter|
-        excluding.include?(filter.field_name)
-      end
+      filters = combine_filters(user_filter.filters_payload(excluding), :and)
```

Comment:
We cleaned this up a bit by renaming a lot of the variables that were confusing. It's still a bit difficult to follow I think, for example in some places we can add nils that ultimately get compacted away again. But hopefully its a bit clearer now what is a query hash and what is an array of queries.